### PR TITLE
Issue #12716: Make GitHub workflow filenames consistent

### LINKED ISSUE #12716: Make scripts and configurations filenames consistent
The `.github/workflows` folder is inconsistent with the filenames inside it. We have some mix of `kebab-case` and `snake_case`.

```
stoyan @ zenbook: ~/open-source/checkstyle/.github/workflows                                                                                                                master [↑]
$ ls                                                                                                                                                                               10:38:44 pm
actionlint.bak                         codeql-analysis.yml           new-milestone-and-issue-in-other-repo.yml  publish_releasenotes_outside.yml  site.yml
bump_license_year.yml                  copy_site_to_sourceforge.yml  no-exception-workflow.yml                  rebase.yml                        update_github_io.yml
bump-version-and-update-milestone.yml  diff_report.yml               no_old_refs.yml                            releasenotes-gen.yml              update_sources.yml
checker-framework.yml                  error-prone.yml               pitest.yml                                 run_link_check.yml                upload-all-jar.yml
check-pr-description.yml               maven_prepare.yml             publish_releasenotes_for_twitter.yml       shellcheck.yml

```

I noticed GitHub uses `kebab-case` in their GitHub Actions documentation, but some people suggest depending on the surrounding software so that would be `PascalCase` in checkstyle's case? For me, it doesn't matter as long as it's consistent.

---
Further edit:
Since we are covering 
> We should use kebab-case for naming of scripts or configs.

we include the scripts in the `.ci` folder as well.

# Folders to cover
- [x] `.ci`
- [x] `.github/workflows`
- [x] `config`

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #12716

---
Before changing the filenames, I ran checkstyle to make sure the check works:
```
[INFO] --- maven-antrun-plugin:3.1.0:run (ant-phase-verify) @ checkstyle ---
[INFO] Executing tasks
[INFO]      [echo] Checkstyle started (checkstyle_checks.xml): 06/02/2023 09:24:15 AM
[INFO] [checkstyle] Running Checkstyle  on 1496 files
[INFO]      [echo] Checkstyle finished (checkstyle_checks.xml) : 06/02/2023 09:25:14 AM
[INFO]      [echo] Checkstyle started (checkstyle_non_main_files_checks.xml): 06/02/2023 09:24:15 AM
[INFO] [checkstyle] Running Checkstyle  on 3535 files
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.ci/idea_inspection.bat:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.ci/idea_inspection.sh:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.ci/no_old_refs.sh:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.ci/sonar_break_build.sh:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/bump_license_year.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/copy_site_to_sourceforge.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/diff_report.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/maven_prepare.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/maven_release_perform.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/no_old_refs.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/publish_releasenotes_for_twitter.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/publish_releasenotes_outside.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/run_link_check.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/update_github_io.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.github/workflows/update_sources.yml:1: All script and configuration files in the '.ci'                     and '.github/workflows' folders should be kebab-case. [regexpOnScriptsAndConfigs]
[INFO] ------------------------------------------------------------------------
```

### CODE REVIEW

File: config/checkstyle_non_main_files_checks.xml
Comment (romani) [inline config/checkstyle_non_main_files_checks.xml:29]: Please put `name="fileExtensions" value="sh, groovy, cmd,`
As next line after folderPatten.

Should folder pattern be just ".ci" folder?

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: config/checkstyle_non_main_files_checks.xml
Comment (rnveach) [reply config/checkstyle_non_main_files_checks.xml:29]: > Should folder pattern be just ".ci" folder?

https://github.com/checkstyle/checkstyle/issues/12716#issue-1570511229
> Make GitHub workflow filenames consistent 
> The .github/workflows folder is inconsistent 

It doesn't look like the issue is about the CI folder. Why are we including the CI folder?

Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml:29]: > We should use kebab-case for naming of scripts or configs.

This is a bit broad now that I see. I suppose this means files in `config` folder as well, right?

Comment (rnveach) [reply config/checkstyle_non_main_files_checks.xml:29]: I think the issue is getting out of hand now and needs to be explicit what it is doing. I wouldn't expect this to cover the `config` folder.

@romani ping

Comment (romani) [reply config/checkstyle_non_main_files_checks.xml:29]: Yes, better to agree on what we fix, and it is completely fine to make several PRs to enforce rule on folders one by one or ... But not like big bang PR.

Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml:29]: I suggest we keep this one with `.github/workflows` folder only as the original issue was. Do you agreee?

Comment (romani) [reply config/checkstyle_non_main_files_checks.xml:29]: Yes

File: config/checkstyle_non_main_files_checks.xml
Comment (romani) [inline config/checkstyle_non_main_files_checks.xml:35]: Violation will be for single file, let's make it compact  , something like `filename should be kebab-case`

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: config/checkstyle_non_main_files_checks.xml
Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml]: Done.

File: config/checkstyle_non_main_files_checks.xml
Comment (romani) [inline config/checkstyle_non_main_files_checks.xml:34]: `Filename should be in kebab-case.`

>then the file name should be in PascalCase.

https://docs.devland.is/technical-overview/adr/0009-naming-files-and-directories

Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml:34]: Alright. Done. I wish Grammarly was avaialbe for IntelliJ Idea so it could spot this :D

**Comment by stoyanK7** [comment]
@romani Which folders/scripts/files specifically would you want to target as well so I open an issue about them.

**Comment by romani** [comment]
We will keep existing issue open, we already reference folders: config, .ci

**Comment by stoyanK7** [comment]
Alright, I will wait for this to get merged before opening PRs for the others.

**Comment by rnveach** [comment]
@stoyanK7 CI in sevntu is failing.
````
[INFO] [checkstyle] Running Checkstyle 10.7.1-SNAPSHOT on 313 files 09:13
[ERROR] [checkstyle] [ERROR] /home/semaphore/checkstyle/.ci-temp/sevntu.checkstyle/.github/workflows/bump_license_year.yml:1: Filename should be kebab-case. [regexpOnScriptsAndConfigs]
````

File: config/checkstyle_non_main_files_checks.xml
Comment (romani) [reply config/checkstyle_non_main_files_checks.xml:34]: Still not done

**Comment by stoyanK7** [comment]
@rnveach I suppose the right course of action is to find a way to limit the check to only `checkstyle/.github/workflow` dir and nothing else?

**Comment by romani** [comment]
To avoid cascade of change in other repositores, let's give I'd to Check and suppress it in all other repositores for now.

**Comment by rnveach** [comment]
I am not seeing why we can't just apply the same fix to sevntu. This is why we have it running in our CI.

**Comment by romani** [comment]
We can , but I am ok to do this step by step.

**Comment by stoyanK7** [comment]
Trying to add suppression in `checkstyle_non_main_files_suppressions.xml` but no success:
```xml
  <module name="RegexpOnFilename">
    <property name="id" value="regexpOnGithubWorkflows"/>
...
```
```xml
  <suppress id="regexpOnGithubWorkflows" files="\.ci\-temp[\\/].*"/>
```
```sh
$ ./.ci/validation.sh no-error-sevntu-checks  
```
```
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.ci-temp/sevntu.checkstyle/.github/workflows/bump_license_year.yml:1: Filename should be in kebab-case. [regexpOnGithubWorkflows]

```
I tried those: 
- `.ci-temp[\\/].*`
 - `\.ci-temp[\\/].*`
 - `.ci\-temp[\\/].*`
 -  `\.ci\-temp[\\/].*`

**Comment by romani** [comment]
Yes, 
Error is on files in other repository
`[ERROR] [checkstyle] [ERROR] /home/semaphore/checkstyle/.ci-temp/sevntu.checkstyle/.github/workflows/bump_license_year.yml:1: Filename should be kebab-case. [regexpOnScriptsAndConfigs]`

We need to send PR to that repo to rename this file before merging this PR. Or send PR to extend suppression file in that repo, also separate PR.

It sounds like renaming in sevntu repo for GitHub action will be quicker to do.

**Comment by romani** [comment]
@stoyanK7 , lets finish this PR before https://github.com/checkstyle/checkstyle/pull/12733/files

**Comment by stoyanK7** [comment]
On it.

**Comment by stoyanK7** [comment]
I will push again after PR in sevntu is merged.

File: config/checkstyle_non_main_files_checks.xml
Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml]: Should be done now. I thought I did the change back then, huh.

**Comment by stoyanK7** [comment]
Done.

**Review by romani — APPROVED** [review]
ok to merge

File: config/checkstyle_non_main_files_checks.xml
Comment (rnveach) [inline config/checkstyle_non_main_files_checks.xml:30]: First parenthesis surrounding `[a-z][a-z0-9]*` isn't needed.

Comment (rnveach) [inline config/checkstyle_non_main_files_checks.xml:30]: We shouldn't end the regular expression with `.*`. You are allowing non-kebab-case characters in, and even if it isn't the real extension.
Example: `kebab.NoTMyKebab_CASE.yml`

I recommend `[a-z]+` as file extensions already covers the extensions we will be matching against.

Comment (rnveach) [inline config/checkstyle_non_main_files_checks.xml:30]: `-` should be sanitized since it is a character as is and we don't want it to accidentally picked up as something else.
`-` => `\-`

**Review by rnveach — CHANGES_REQUESTED** [review]

File: config/checkstyle_non_main_files_checks.xml
Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml]: Done.

Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml]: I agree, done.

Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml]: Done.

**Review by rnveach — APPROVED** [review]

**Pull request merged**
